codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/java/model/second/JavaObjectTypeRenderer.kt

```diff
@@ -0,0 +1,62 @@
+package io.vrap.codegen.languages.java.model.second
+
+import io.vrap.codegen.languages.extensions.hasSubtypes
+import io.vrap.codegen.languages.java.extensions.EObjectTypeExtensions
+import io.vrap.codegen.languages.java.extensions.ObjectTypeExtensions
+import io.vrap.codegen.languages.java.extensions.simpleName
+import io.vrap.rmf.codegen.rendring.ObjectTypeRenderer
+import io.vrap.rmf.raml.model.types.ArrayType
+import io.vrap.rmf.raml.model.types.ObjectType
+import io.vrap.rmf.raml.model.types.Property
+import io.vrap.rmf.raml.model.types.util.TypesSwitch
+import org.eclipse.emf.ecore.EObject
+
+abstract class JavaObjectTypeRenderer : ObjectTypeExtensions, EObjectTypeExtensions, ObjectTypeRenderer {
+
+    fun ObjectType.subTypesAnnotations(): String {
+        return if (this.hasSubtypes())
+            """
+            |@JsonSubTypes({
```

Comment:
If we decide to only have this annotations on interfaces - which I suggested above - we should move this extension method to the `JavaObjecttypeInterfaceRenderer`.